src/sql-parser/src/parser.rs

```diff
@@ -1160,6 +1167,92 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    fn parse_operator_expr(&mut self) -> Result<Token, ParserError> {
+        self.expect_token(&Token::LParen)?;
+        let tok = self
+            .next_token()
+            .ok_or_else(|| self.error(self.peek_prev_pos(), "Unexpected EOF".to_string()))?;
```

Comment:
Although I see this existing pattern in `parse_prefix`, I wonder if we can use the other pattern which would allow for the removal of the EOF check:

```
match self.next_token() {
  Some(X) => ..
  t => self.expected(self.peek_prev_pos(), "..", t)?
```